### PR TITLE
Merge breakpad upstream

### PR DESCRIPTION
**Pull request by MarijnS95**
Probably more good reasons than just this, but my Arch install upgraded to GCC 13 yesterday and that triggers:

```c
  cargo:warning=In file included from breakpad/src/client/linux/handler/minidump_descriptor.cc:32:
  cargo:warning=breakpad/src/client/linux/handler/minidump_descriptor.h:115:3: error: ‘uintptr_t’ does not name a type
  cargo:warning=  115 |   uintptr_t address_within_principal_mapping() const {
  cargo:warning=      |   ^~~~~~~~~
  cargo:warning=breakpad/src/client/linux/handler/minidump_descriptor.h:40:1: note: ‘uintptr_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
  cargo:warning=   39 | #include "common/using_std_string.h"
  cargo:warning=  +++ |+#include <cstdint>
  cargo:warning=   40 |
```

Upstream already has a solution at https://github.com/getsentry/breakpad/commit/7ea7ded187b4d739239f3ab7082fcd5a2ccc1eaa, so we just merge in main completely.

I'm not sure if we should also go through the sentry fork, or do a clean rebase.  Alternatively we can also pick just the patch that fixes the build?

### CODE REVIEW

File: src/processor/stackwalker.cc [hunk 1]
```diff
@@ -321,6 +327,30 @@ Stackwalker* Stackwalker::StackwalkerForCPU(
...
+    {
+      const MDRawContextRISCV64* context_riscv64 = context->GetContextRISCV64();
+      // memory = GetActualStackMemory(context_riscv64->__gregs[MD_CONTEXT_RISCV_REG_SP],
+      //                               memory, memory_list);
+      cpu_stackwalker = new StackwalkerRISCV64(system_info,
+                                               context_riscv64,
+                                               memory, modules,
+                                               frame_symbolizer);
+      break;
+    }
```
Comment (MarijnS95) [inline src/processor/stackwalker.cc:353]: This is the only conflict (out of two) I could not resolve: haven't figured out where the stack-pointer register is.


https://github.com/getsentry/breakpad/pull/1

**Comment by MarijnS95** [comment]
> I'm not sure if we should also go through the sentry fork, or do a clean rebase

Sentry's `handler` (default) branch seems to be actively updated with Google's `main`, but misses some of the changes that were picked to Embark's fork, or so it seems.

**Comment by MarijnS95** [comment]
@Jake-Shadle how do you want to proceed here and in https://github.com/EmbarkStudios/sentry-contrib-rust/pull/24?

**Comment by MarijnS95** [comment]
It's ever so slightly painful that the *squash* merge now does not track when we last integrated the `upstream/main` branch, resulting in all kinds of conflicts when we merge again :(

(And the "x commits behind upstream" text is also wrong - and how about Sentry's own fork of Google Breakpad? :cold_sweat:)

**Comment by Jake-Shadle** [comment]
Oops.

**Comment by MarijnS95** [comment]
> Oops.

https://github.com/EmbarkStudios/breakpad/compare/main..MarijnS95:breakpad:merge-upstream contains the remaining changes of merging `Google/breakpad:main` into this PR branch before it was squash-merged into this repo, as well as merging in `getsentry/breakpad:handler` which doesn't diverge much from Embark's fork which was based on it initially, but does diverge quite a bit from Google's upstream. They do merge in Google's upstream quite often though, so we might also "just" rebase on top of their `handler` branch instead for better tracking?

I can help address that if you want :)

**Comment by Jake-Shadle** [comment]
Sure.

**Comment by MarijnS95** [comment]
@Jake-Shadle done, the branch is identical to this original branch with the latest `getsentry/handler` branch merged in (as shared 2 weeks ago):

https://github.com/MarijnS95/breakpad/compare/rebase-upstream..merge-upstream

**Comment by MarijnS95** [comment]
@Jake-Shadle do you still intend to go the rebase-route?

**Comment by Jake-Shadle** [comment]
Yes, need a PR

**Comment by MarijnS95** [comment]
@Jake-Shadle sure, if you reset this repo to the upstream branch first, make sure you then rebase-merge instead of squash-merge it :grimacing:

**Comment by MarijnS95** [comment]
Done in #3 :)

**Pull request merged**